packages/admin/cms-admin/src/generator/future/generator.ts

```diff
@@ -22,31 +25,42 @@ export type FormFieldConfig<T> = (
     | { type: "staticSelect"; values?: string[] }
     | { type: "asyncSelect"; values?: string[] }
     | { type: "block"; block: BlockReference }
-) & { name: keyof T; label?: string; required?: boolean };
+) & { name: string; label?: string; required?: boolean };
+export type FormFieldConfig<T extends GeneratorEntity> = FormFieldConfigInternal & { name: Leaves<T> | Paths<T> };
 
-export type FormConfig<T extends { __typename?: string }> = {
+export type FormConfigInternal = {
     type: "form";
-    gqlType: T["__typename"];
+    gqlType: string;
     fragmentName?: string;
-    fields: FormFieldConfig<T>[];
     title?: string;
+    fields: FormFieldConfigInternal[];
+};
+export type FormConfig<T extends GeneratorEntity> = FormConfigInternal & {
+    gqlType: T["__typename"];
+    fields: FormFieldConfig<T>[];
 };
 
 export type TabsConfig = { type: "tabs"; tabs: { name: string; content: GeneratorConfig }[] };
 
-export type GridColumnConfig<T> = (
+export type GridColumnConfigInternal = (
     | { type: "text" }
     | { type: "number" }
     | { type: "boolean" }
     | { type: "date" }
     | { type: "dateTime" }
     | { type: "staticSelect"; values?: string[] }
     | { type: "block"; block: BlockReference }
-) & { name: keyof T; headerName?: string; width?: number };
-export type GridConfig<T extends { __typename?: string }> = {
+) & { name: string; headerName?: string; width?: number };
+export type GridColumnConfig<T extends GeneratorEntity> = GridColumnConfigInternal & { name: Leaves<T> | Paths<T> };
+
+export type GridConfigInternal = {
     type: "grid";
-    gqlType: T["__typename"];
+    gqlType: string;
     fragmentName?: string;
+    columns: GridColumnConfigInternal[];
+};
+export type GridConfig<T extends GeneratorEntity> = GridConfigInternal & {
```

Comment:
What it the motivation for  GridConfigInternal?

Reply:
To be consistent with FormConfig and simplify usage in admin-generator code, as there is no need for GridConfig<any> anywhere.

Reply:
I meant because I added FormConfigInternal I also added GridConfigInternal, for them to be consistent. And I think this is a good idea because we don't know the type in the generic-brackets and can now remove all of the ignore-any comments. and for now this just simplifies the type intern and the additional info is currently not needed.

Reply:
IMO ...ConfigInternal a bad name for something that's exported and used in many places. If it's only about removing the ESLint disables, you could also add a default value to the generic 🤷🏼‍♂️

Reply:
I understand now your motivation and it is indeed annoying to always have to disable eslint. Alternative idea: `type AnyGridConfig = GridConfig<any>;`

And this would be a perfect candidate for it's own PR as it is completely unrelated to nested fields. You can still do that.

Reply:
> I understand now your motivation and it is indeed annoying to always have to disable eslint.

to be clear: this was just a handy side effect. the main reason for ConfigInternal was typescript complaining in different places having a hard time to infer the type of `name` after defining it as `Path | Leave`. And in my opinion it's much cleaner to redefine the base-type instead of adding types all over the code like for `array.map<string>`.

Maybe using your `UsableFieldsInner` can solve the root issue.